File: Splitwise/Split.java
public class Split {
    User user;
    double amountOwe;
    double percentage = 0.0;

    public Split(User user, double amountOwe) {
        this.user = user;
        this.amountOwe = amountOwe;
        this.percentage = 0.0;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public double getAmountOwe() {
        return amountOwe;
    }

    public void setAmountOwe(double amountOwe) {
        this.amountOwe = amountOwe;
    }

    public double getPercentage() {
        return percentage;
    }
}

File: Go/SolidPrinciples/SplitwiseApp/Models/Split.go
package Models

type Split struct {
	User       *User
	Amount     float64
	Percentage float64
}

func NewSplit(user *User, amount float64) *Split {
	return &Split{user, amount, 0}
}

func (s *Split) GetAmount() float64 {
	return s.Amount
}

func (s *Split) GetPercentage() float64 {
	return s.Percentage
}

func (s *Split) GetUser() *User {
	return s.User
}
